### DISCUSSION TITLE
specific lookback setting in time series

### DISCUSSION THREAD

**Discussion by SyedKumailHussainNaqvi** [autogluon/autogluon#3922] [Q&A]
How can i set specific lookback (window size or sequence length) in autogluon time series  .
I am working on PV power forecasting, and I would like to set the lookback (window size or sequence length) to 36. Additionally, my dataset has a time step of 5 minutes and and the predict length(horizon) is 9.

Could someone kindly guide me on how to configure Autogluon for this specific lookback setting? If there are any examples or code snippets available, that would be greatly appreciated.

train_data = TimeSeriesDataFrame.from_data_frame(
df,
id_column="id",
timestamp_column="timestamp"
)
predictor = TimeSeriesPredictor(
prediction_length=9,
path="autogluon-m4-hourly",
target="Active_Power",
freq='5min',

eval_metric="MASE",
)
predictor.fit(
train_data,
presets="best_quality",
time_limit=900,
)

**Comment by shchur** [comment]
Hi, to make that I understand your question correctly, by "setting the lookback to X" you mean restricting all models such that they don't use more than X previous observations when making the forecast?

If you meant something different, could you provide some examples of how this is handled in other forecasting packages, or maybe a small example with a dummy time series like `[1, 2, 3, 4, 5, 6, 7, 8, 9]`?
